diff --git a/.gitconfig b/.gitconfig
index 7e3e8d4..f2bffe8 100644
--- a/.gitconfig
+++ b/.gitconfig
@@ -1,7 +1,9 @@
 [core]
     editor = vim
 [user]
-    email = gravell.contact@gmail.com
-	name = gravell
+    email = pcharkie@amazon.pl
+	name = Pawel Charkiewicz
 [diff]
 	tool = vimdiff
+[color]
+	ui = auto
diff --git a/.vimrc b/.vimrc
index 5983e94..568936c 100644
--- a/.vimrc
+++ b/.vimrc
@@ -2,6 +2,13 @@
 " - Fix auto jumping to first cscope result
 " - Add manual search for cscope plugin (manual typing)
 " - cleanup
+" - Plugin fixes:
+" -- QuickrPreview:
+" --- quickr_preview_exit_on_enter: cclose; lclose;
+" -- Cscove:
+" --- Exculde 'x' action from CscopeFind
+" - Add proejct specific settings (i.e. exclude paths for CtrlP and CScope)
+" - Add dictionary
 
 "Set - as default leader
 let mapleader="-"
@@ -68,15 +75,18 @@ nnoremap <leader>so :so $MYVIMRC<CR>
 nnoremap <leader>se :e $MYVIMRC<CR>
 
 " Enable magic
-nnoremap / /\v
-vnoremap / /\v
-cnoremap s/ s/\v
-cnoremap g/ g/\v
+" nnoremap / /\v
+" vnoremap / /\v
+" cnoremap s/ s/\v
+" cnoremap g/ g/\v
 
 " * and # highlights but doesn't jump to next by default
 " nnoremap * :keepjumps normal *``<CR>
 " nnoremap # :keepjumps normal #``<CR>
 
+" Search for string within other strings
+nnoremap <silent> g* :let @/=expand('<cword>') <bar> set hls <cr>
+
 " Current line highlighting and line numbering
 set number
 set relativenumber
@@ -85,7 +95,7 @@ set cursorline
 " Disable swap files
 set noswapfile
 
-" When next buffer is openedthe currently modified one goes into background
+" When next buffer is opened the currently modified one goes into background
 set hidden
 
 " Enable mouse
@@ -149,6 +159,7 @@ let g:airline_powerline_fonts=1
 let g:airline_theme='monokai'
 
 " Ctrlp options
+let g:ctrlp_max_files=0
 let g:ctrlp_map='<c-p>'
 let g:ctrlp_match_window = 'results:100'
 let g:ctrlp_working_path_mode='a'
@@ -156,6 +167,9 @@ let g:ctrlp_custom_ignore={
     \'dir': '\v[\/]\.(git|hg|svn)$',
     \'file': '\v\.(o|disasm|hex|readelf|ioc|bin|exe|sc|dll)$'
     \}
+let g:ctrlp_by_filename=1
+let g:ctrlp_regexp=1
+let g:ctrlp_lazy_update=1
 
 " Nerdcommenter options
 let g:NERDCustomDelimiters = { 'c': { 'left': '//'} }
@@ -184,6 +198,8 @@ command! W w
 
 " NerdTree options
 nnoremap <silent> <leader>ne :NERDTreeToggle<CR>
+nnoremap <silent> <leader>nf :NERDTreeFind<cr>
+nnoremap <silent> <leader>nr :NERDTreeRefreshRoot<cr>
 
 " Tagbar options
 nnoremap <silent> <leader>t :TagbarToggle<CR>
@@ -206,7 +222,9 @@ nnoremap <silent> <c-w>w :ZoomWin<CR>
 " Cscove options
 let g:cscope_silent = 1
 let g:cscope_auto_update = 0
-let g:cscope_ignored_dir ="(build_scripts\|docs\|fpga\|.git\|githooks\|outputs\|rom\|signed\|simics-output\|tools\|versions)"
+" Example of ignore string:
+" let g:cscope_ignored_dir ="(folder1\|.folder2\|.folder3)"
+let g:cscope_ignored_dir ="(prebuilds\|.repo\|.git\|githooks)"
 
 function! RefreshCscopeDatabase()
     :silent exec ":CscopeClear"
diff --git a/mc/ini b/mc/ini
index 2dfd733..b61cff1 100644
--- a/mc/ini
+++ b/mc/ini
@@ -1,81 +1,81 @@
 [Midnight-Commander]
-verbose=true
+verbose=0
 pause_after_run=1
-shell_patterns=true
-auto_save_setup=true
-preallocate_space=false
-auto_menu=false
-use_internal_view=true
-use_internal_edit=true
-clear_before_exec=true
-confirm_delete=true
-confirm_overwrite=true
-confirm_execute=false
-confirm_history_cleanup=true
-confirm_exit=true
-confirm_directory_hotlist_delete=true
-safe_delete=false
+shell_patterns=0
+auto_save_setup=1
+preallocate_space=0
+auto_menu=0
+use_internal_view=0
+use_internal_edit=0
+clear_before_exec=0
+confirm_delete=0
+confirm_overwrite=0
+confirm_execute=0
+confirm_history_cleanup=0
+confirm_exit=0
+confirm_directory_hotlist_delete=0
+safe_delete=0
 mouse_repeat_rate=100
 double_click_speed=250
-use_8th_bit_as_meta=false
-confirm_view_dir=false
-mouse_move_pages_viewer=true
-mouse_close_dialog=false
-fast_refresh=false
-drop_menus=false
-wrap_mode=true
-old_esc_mode=false
+use_8th_bit_as_meta=0
+confirm_view_dir=0
+mouse_move_pages_viewer=0
+mouse_close_dialog=0
+fast_refresh=0
+drop_menus=0
+wrap_mode=0
+old_esc_mode=0
 old_esc_mode_timeout=1000000
-cd_symlinks=true
-show_all_if_ambiguous=false
+cd_symlinks=0
+show_all_if_ambiguous=0
 max_dirt_limit=10
-use_file_to_guess_type=true
-alternate_plus_minus=false
-only_leading_plus_minus=true
-show_output_starts_shell=false
-xtree_mode=false
+use_file_to_guess_type=0
+alternate_plus_minus=0
+only_leading_plus_minus=0
+show_output_starts_shell=0
+xtree_mode=0
 num_history_items_recorded=60
-file_op_compute_totals=true
-classic_progressbar=true
+file_op_compute_totals=0
+classic_progressbar=0
 vfs_timeout=60
 ftpfs_directory_timeout=900
-use_netrc=true
+use_netrc=0
 ftpfs_retry_seconds=30
-ftpfs_always_use_proxy=false
-ftpfs_use_passive_connections=true
-ftpfs_use_passive_connections_over_proxy=false
-ftpfs_use_unix_list_options=true
-ftpfs_first_cd_then_ls=true
+ftpfs_always_use_proxy=0
+ftpfs_use_passive_connections=0
+ftpfs_use_passive_connections_over_proxy=0
+ftpfs_use_unix_list_options=0
+ftpfs_first_cd_then_ls=0
 fish_directory_timeout=900
 editor_tab_spacing=8
 editor_word_wrap_line_length=72
-editor_fill_tabs_with_spaces=false
-editor_return_does_auto_indent=false
-editor_backspace_through_tabs=false
-editor_fake_half_tabs=true
+editor_fill_tabs_with_spaces=0
+editor_return_does_auto_indent=0
+editor_backspace_through_tabs=0
+editor_fake_half_tabs=0
 editor_option_save_mode=0
-editor_option_save_position=true
-editor_option_auto_para_formatting=false
-editor_option_typewriter_wrap=false
-editor_edit_confirm_save=true
-editor_syntax_highlighting=true
-editor_persistent_selections=true
+editor_option_save_position=0
+editor_option_auto_para_formatting=0
+editor_option_typewriter_wrap=0
+editor_edit_confirm_save=0
+editor_syntax_highlighting=0
+editor_persistent_selections=0
 editor_drop_selection_on_copy=true
-editor_cursor_beyond_eol=false
-editor_cursor_after_inserted_block=false
-editor_visible_tabs=true
-editor_visible_spaces=true
-editor_line_state=false
-editor_simple_statusbar=false
-editor_check_new_line=false
-editor_show_right_margin=false
-editor_group_undo=true
+editor_cursor_beyond_eol=0
+editor_cursor_after_inserted_block=0
+editor_visible_tabs=0
+editor_visible_spaces=0
+editor_line_state=0
+editor_simple_statusbar=0
+editor_check_new_line=0
+editor_show_right_margin=0
+editor_group_undo=0
 editor_state_full_filename=true
 editor_ask_filename_before_edit=false
-nice_rotating_dash=true
-mcview_remember_file_position=false
-auto_fill_mkdir_name=true
-copymove_persistent_attr=true
+nice_rotating_dash=0
+mcview_remember_file_position=0
+auto_fill_mkdir_name=0
+copymove_persistent_attr=0
 editor_backup_extension=~
 editor_filesize_threshold=64M
 editor_stop_format_chars=-+*\\,.;:&>
@@ -87,20 +87,24 @@ filepos_max_saved_entries=1024
 
 safe_overwrite=false
 
+select_flags=6
+keymap=
+
 [Layout]
 message_visible=0
-keybar_visible=1
-xterm_title=1
+keybar_visible=0
+xterm_title=0
 output_lines=0
-command_prompt=1
-menubar_visible=1
-free_space=1
+command_prompt=0
+menubar_visible=0
+free_space=0
 horizontal_split=0
 vertical_equal=1
-left_panel_size=60
-horizontal_equal=1
+horizontal_equal=0
 top_panel_size=1
 
+left_panel_size=142
+
 [Misc]
 timeformat_recent=%H:%M %d.%m.%Y
 timeformat_old=%H:%M %d.%m.%Y
@@ -134,7 +138,7 @@ fast_reload=false
 fast_reload_msg_shown=false
 mark_moves_down=true
 reverse_files_only=true
-auto_save_setup_panels=false
+auto_save_setup_panels=true
 navigate_with_arrows=false
 panel_scroll_pages=true
 mouse_move_pages=true
diff --git a/mc/panels.ini b/mc/panels.ini
index 596f1e6..3b2e776 100644
--- a/mc/panels.ini
+++ b/mc/panels.ini
@@ -1,7 +1,7 @@
 [New Left Panel]
 display=listing
 reverse=0
-case_sensitive=1
+case_sensitive=0
 exec_first=0
 sort_order=name
 list_mode=user
@@ -11,12 +11,14 @@ user_status0=half type name | size | perm
 user_status1=half type name | size | perm
 user_status2=half type name | size | perm
 user_status3=half type name | size | perm | mtime
-user_mini_status=1
+user_mini_status=0
+
+list_format=user
 
 [New Right Panel]
 display=listing
 reverse=0
-case_sensitive=1
+case_sensitive=0
 exec_first=0
 sort_order=name
 list_mode=user
@@ -26,4 +28,9 @@ user_status0=half type name | size | perm
 user_status1=half type name | size | perm
 user_status2=half type name | size | perm
 user_status3=half type name | size | perm | mtime
-user_mini_status=1
+user_mini_status=0
+list_format=user
+
+[Dirs]
+current_is_left=true
+other_dir=/home/pcharkie/repos/wwr
